msrest/paging.py

```diff
@@ -73,22 +70,38 @@ def raw(self):
         return raw
 
     def get(self, url):
-        """Get arbitrary page.
+        """Get an arbitrary page.
+
+        This resets the iterator and then fully consumes it to return the
+        specified page.
 
         :param str url: URL to arbitrary page results.
         """
+        self.reset()
         self.next_link = url
-        return self.next()
+        return list(self)
```

Comment:
Why wouldn't it work? This is simply exhausting the iterator so if a "page 3" would have come through the iterator then this will as well.

And are you adding a test for this?